### PR TITLE
Add nodeAffinity to run only on master

### PR DESCRIPTION
**Pull request by marioferh**
Add nodeAffinity to run only on master
Signed-off-by: Mario Fernandez <mariofer@redhat.com>

### CODE REVIEW

**Comment by fromanirh** [comment]
at glance the change is correct, but misplaced. We need to make sure this snipped is preserved once we do `make generate` and/or `make manifests`.

**Comment by cynepco3hahue** [comment]
I think you need to set change here https://github.com/openshift-kni/numaresources-operator/blob/main/config/manager/manager.yaml and re-generate manifests.

**Comment by fromanirh** [comment]
> I think you need to set change here https://github.com/openshift-kni/numaresources-operator/blob/main/config/manager/manager.yaml and re-generate manifests.

yes, this seems the right place. Thanks for checking quickly, @cynepco3hahue

**Comment by marioferh** [comment]
> I think you need to set change here https://github.com/openshift-kni/numaresources-operator/blob/main/config/manager/manager.yaml and re-generate manifests.

Done, thanks

File: bundle/metadata/annotations.yaml [hunk 1]
```diff
@@ -5,7 +5,6 @@ annotations:
   operators.operatorframework.io.bundle.metadata.v1: metadata/
   operators.operatorframework.io.bundle.package.v1: numaresources-operator
   operators.operatorframework.io.bundle.channels.v1: alpha
-  operators.operatorframework.io.bundle.channel.default.v1: alpha
```
Comment (fromanirh) [inline bundle/metadata/annotations.yaml:8]: why was this removed?

Comment (cynepco3hahue) [inline bundle/metadata/annotations.yaml:8]: @fromanirh I remember you added it, probably we should add some parameter to avoid override by `make bundle`

Comment (marioferh) [reply bundle/metadata/annotations.yaml:8]: override by make bundle

Comment (fromanirh) [reply bundle/metadata/annotations.yaml:8]: good point, need to check why is being deleted

Comment (fromanirh) [reply bundle/metadata/annotations.yaml:8]: handled here: https://github.com/openshift-kni/numaresources-operator/pull/75 
not blocking for this PR

File: bundle/manifests/numaresources-operator.clusterserviceversion.yaml [hunk 2]
```diff
@@ -257,9 +264,6 @@ spec:
                   initialDelaySeconds: 5
                   periodSeconds: 10
                 resources:
-                  limits:
-                    cpu: 200m
-                    memory: 100Mi
```
Comment (fromanirh) [inline bundle/manifests/numaresources-operator.clusterserviceversion.yaml:262]: I like this, the operator pod should really not have limits until/unless we have a good rationale - and proof the limits will not backfire. And since AFAIK we don't have this data, better not to guess. The only problem is, this should happen more explicitely :\

Comment (marioferh) [reply bundle/manifests/numaresources-operator.clusterserviceversion.yaml:262]: This come from here: https://github.com/openshift-kni/numaresources-operator/pull/71
Was generated after make bundle

Comment (fromanirh) [reply bundle/manifests/numaresources-operator.clusterserviceversion.yaml:262]: yup. For the reasons above, it's good we remove the limits. But we should remove the limitis explicitely, with a separate commit/PR and proper explanation, and not causally by incident on an apparently unrelated PR which happened to need to run `make bundle`.

Comment (marioferh) [reply bundle/manifests/numaresources-operator.clusterserviceversion.yaml:262]: Ok, done: https://github.com/openshift-kni/numaresources-operator/pull/77

Comment (marioferh) [reply bundle/manifests/numaresources-operator.clusterserviceversion.yaml]: Rebase and fix it

File: bundle/metadata/annotations.yaml (hunk 1 above)
Comment (marioferh) [reply bundle/metadata/annotations.yaml]: Fixed it

**Comment by fromanirh** [comment]
besides the conflict, the test failures look real

**Comment by marioferh** [comment]
> besides the conflict, the test failures look real

It is, investigating
